### PR TITLE
[FIX] update developer installation instruction

### LINKED ISSUE #3376: Developer installation instruction is outdated
We don't use requirement files anymore so this should be update to.

```
python -m pip install -e .[dev]
```

### Affected pages
https://nilearn.github.io/stable/introduction.html#installation

### PR DESCRIPTION
**Pull request by htwangtw**
Closes #3376.

We have moved on from requirements files to `setup.cfg` and the documentation here needs an update!

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: doc/introduction.rst [hunk 1]
```diff
@@ -149,8 +149,7 @@ Installing ``nilearn``
 
       .. code-block::
 
-         python -m pip install -r requirements-dev.txt
-         python -m pip install -e .
+         python -m pip install -e .[dev]
```
Comment (ymzayek) [inline doc/introduction.rst:152]:
```suggestion
         python -m pip install -e '.[dev]'
```

**Review by ymzayek — APPROVED** [review]

**Review by NicolasGensollen — APPROVED** [review]
Thanks @htwangtw !

**Comment by ymzayek** [comment]
Merging

**Pull request merged**